### PR TITLE
Feat/add testing

### PR DESCRIPTION
**Pull request by AndreeaLupu2000**
### Frontend

File | Component Tested | Test Focus | Type
-- | -- | -- | --
RegisterView.test.tsx | <RegisterView /> | Form submission, API interaction, error handling | Integration
LoginView.test.tsx | <LoginView /> | UI validation, API login flow, session storage, auth failures | Integration
PhotoView.test.tsx | <PhotoView /> | Photo capture & send actions using mocked custom hooks | Integration
VideoRecordingView.test.tsx | <VideoRecordingView /> | Start/stop video recording, sending preview via mocked hooks | Integration
MicrophoneView.test.tsx | <MicrophoneView /> | Voice interaction flow, start/stop/send/clear conversation | Integration
ticketService.test.ts | ticketService | Ticket-related API methods and error handling | Unit
userService.test.ts | userService | User data retrieval, API interactions, role-based filters | Unit
mediaService.test.ts | mediaService | Media upload/download logic (e.g., photo/video) | Unit
voiceService.test.ts | voiceService | Audio processing, storage, and conversation logic | Unit

### GenAI

Test Type | Files Involved | Notes
-- | -- | --
Unit Tests | test_utils.py | Focus on individual utilities with mocking.
Integration Tests | test_gemini_photo.py, test_utils.py | Likely test client interactions and environment.
Test Infra / Fixtures | conftest.py | Typically holds reusable fixtures for pytest.

### CODE REVIEW

File: client/frontend/src/test/components/RegisterView.test.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        name: 'newuser',
+        password: 'password123',
+        role: 'WORKER'
+      });
+    });
+  });
+
+  it('should handle registration failure with existing user', async () => {
+    mockAxios.onPost('http://localhost:8030/auth/register')
+      .reply(409, { message: 'User already exists' });
```
Comment (rodrigoflx) [inline client/frontend/src/test/components/RegisterView.test.tsx:58]: This is not exactly the error code returned in this situation, but I also didn't finish the API specs in time.

**Review by rodrigoflx — APPROVED** [review]
There are some small inconsistencies related to the expected API responses (see comment above) but they are completely unimportant for the functioning of the code, so we can ignore it. 

Technically, these are not integration tests (every API was mocked) but it's as important to write these tests as integration ones! 

Overall well written and covers all common cases and expected edge cases.

**Comment by AndreeaLupu2000** [comment]
I can update the API responses but I don't know where I can find them

**Comment by rodrigoflx** [comment]
> I can update the API responses but I don't know where I can find them

Let‘s just leave it like this first
